sideboard/config.py

```diff
@@ -9,14 +10,72 @@
 from validate import Validator
 
 
+_defaults_re = re.compile(r'(.+)-defaults(\.\w+)$')
+
+
+def uniquify(x):
+    """
+    Returns an ordered copy of `x` with duplicate items removed.
+
+    >>> uniquify(['a', 'b', 'a', 'c', 'a', 'd', 'a', 'e'])
+    ['a', 'b', 'c', 'd', 'e']
+
+    """
+    seen = set()
+    return [i for i in x if i not in seen and not seen.add(i)]
```

Comment:
A few notes about this function:

1. It really belongs in `sideboard.lib._util`, but trying to import it from there causes a circular import error (even if I import it from inside a function).

2. I usually use "ordered" to mean the the order of elements is relevant and preserved, and "sorted" to mean in ascending or descending order. That being said this docstring is totally ambiguous and needs a rewrite.

3. It doesn't need to be a list comprehension at all, and would be clearer in long form as described above.

4. In my original implementation I **DID** call `listify` on the argument! But then I thought, if someone passes in an argument that is not already "listy" we should probably error on that. For example, if someone accidentally passes in `None`, then we would return `[None]`, which I don't think is what we intend to do. So, for my perspective, if that's really what someone wants, then they can `listify` the argument before passing it to `uniquify`.

5. I used `x` as the parameter name because that's what we also use for `listify` and `is_listy`, but then I was like, "what variable am I gonna use for the items? Oh well, `i` it is!"